src/platform_impl/windows/dark_mode.rs

```diff
@@ -160,40 +165,31 @@ pub fn allow_dark_mode_for_window(hwnd: HWND, is_dark_mode: bool) {
   }
 }
 
-type SetWindowCompositionAttribute =
-  unsafe extern "system" fn(HWND, *mut WINDOWCOMPOSITIONATTRIBDATA) -> BOOL;
-static SET_WINDOW_COMPOSITION_ATTRIBUTE: Lazy<Option<SetWindowCompositionAttribute>> =
-  Lazy::new(|| get_function!("user32.dll", SetWindowCompositionAttribute));
-
-type WINDOWCOMPOSITIONATTRIB = u32;
-const WCA_USEDARKMODECOLORS: WINDOWCOMPOSITIONATTRIB = 26;
-#[repr(C)]
-struct WINDOWCOMPOSITIONATTRIBDATA {
-  Attrib: WINDOWCOMPOSITIONATTRIB,
-  pvData: *mut c_void,
-  cbData: usize,
-}
-
-fn refresh_titlebar_theme_color(hwnd: HWND, is_dark_mode: bool) {
-  // SetWindowCompositionAttribute needs a bigbool (i32), not bool.
-  let mut is_dark_mode_bigbool: i32 = is_dark_mode.into();
-
+fn refresh_titlebar_theme_color(hwnd: HWND, is_dark_mode: bool, redraw_title_bar: bool) {
   if let Some(ver) = *WIN10_BUILD_VERSION {
     if ver < 18362 {
+      let mut is_dark_mode_bigbool: i32 = is_dark_mode.into();
       unsafe {
         let _ = SetPropW(
           hwnd,
           w!("UseImmersiveDarkModeColors"),
           HANDLE(&mut is_dark_mode_bigbool as *mut _ as _),
         );
       }
-    } else if let Some(set_window_composition_attribute) = *SET_WINDOW_COMPOSITION_ATTRIBUTE {
-      let mut data = WINDOWCOMPOSITIONATTRIBDATA {
-        Attrib: WCA_USEDARKMODECOLORS,
-        pvData: &mut is_dark_mode_bigbool as *mut _ as _,
-        cbData: std::mem::size_of_val(&is_dark_mode_bigbool) as _,
-      };
-      let _ = unsafe { set_window_composition_attribute(hwnd, &mut data as *mut _) };
+    } else {
+      let dark_mode = BOOL::from(is_dark_mode);
+      unsafe {
+        let _ = DwmSetWindowAttribute(
+          hwnd,
+          DWMWA_USE_IMMERSIVE_DARK_MODE,
```

Comment:
The official docs on this is not correct, this is available since way before that, I don't have the device to test if it works on all versions after windows 10 build 18362+ though

Reply:
I'd rather be safe than sorry I guess. Unlick `SetWindowCompositionAttribue` function, there is no way to reliably check if `DWMWA_USE_IMMERSIVE_DARK_MODE` is supported by the OS and it could lead to undefined behavior or not working at all.

Reply:
Are there any docs about `WCA_USEDARKMODECOLORS`? I can't really find any, also `SetWindowCompositionAttribute` should always be there according to https://learn.microsoft.com/en-us/windows/win32/dwm/setwindowcompositionattribute this is there from windows 7, checking if this thing exists doesn't do anything for us I don't think

Reply:
> checking if this thing exists doesn't do anything for us I don't think

hmm, I guess the check here then is whether we failed to load it from `user32.dll` or not. Just a misunderstanding on my part then.

Anyways, `WCA_USEDARKMODECOLORS` is not documented, but it exists in older Windows 10 versions, and can be used as a fallback for older versions. 

Since the documentation, says `DWMWA_USE_IMMERSIVE_DARK_MODE` is supported in `Windows 11 Build 22000` and later, I think we should use it as documented. 

We can use `DWMWA_USE_IMMERSIVE_DARK_MODE` for versions older than that but we should test it and confirm first and keep as default fallback `WCA_USEDARKMODECOLORS`.

Reply:
`WCA_USEDARKMODECOLORS` is not documented, but it exists in older Windows 10 versions, and can be used as a fallback for older versions. 

Since the documentation, says `DWMWA_USE_IMMERSIVE_DARK_MODE` is supported in `Windows 11 Build 22000` and later, I think we should use it as documented. 

We can use `DWMWA_USE_IMMERSIVE_DARK_MODE` for versions older than that but we should test it and confirm first and keep as default fallback `WCA_USEDARKMODECOLORS`.

Reply:
I believe `DWMWA_USE_IMMERSIVE_DARK_MODE` works on older versions, I can find a lot of references in a google search about that, but I can't find much about `WCA_USEDARKMODECOLORS`

The docs was pointing to it once but later removed saying this shouldn't be documented that way https://github.com/MicrosoftDocs/sdk-api/pull/966/files

Reply:
I guess that's not to cause confusion because the value of `DWMWA_USE_IMMERSIVE_DARK_MODE` is different in different operating systems, however `windows-rs` and `windows-sys` are generated on newer versions of Windows and it will always be `20` https://docs.rs/windows-sys/latest/windows_sys/Win32/Graphics/Dwm/constant.DWMWA_USE_IMMERSIVE_DARK_MODE.html.

So based on the documentation you provided we should define our own constant `DWMWA_USE_IMMERSIVE_DARK_MODE_BEFORE_20H1 = 19` on lower versions and use that instead.

Reply:
Will do

Reply:
Do you know if `UseImmersiveDarkModeColors` even works on older versions? we currently use it as a fallback on version < 18362

Reply:
I don't remember ever testing it tbh, I think it was something that winit team has implemented, However versions less than 18362 is super old, let's just keep it as is using `UseImmersiveDarkModeColors`.

Reply:
I changed it to < 17763 since `DWMWINDOWATTRIBUTE(19)` should work between 17763 and 18985, keeping it now then